### DISCUSSION TITLE
WS2812b LED's

### DISCUSSION THREAD

**Discussion by edwardker** [talbotmcinnis/dcs-bios-arduino-library#89] [Q&A]
Hi, I would like to program an arduino for warninglights (Hornet EWI panels) using WS2812B LEDs. So the arduino has to receive data from the game and pass that on to the LEDs. I have no clue how to start with this. Could someone help me out? Appreciate any help.

**Comment by charliefoxtwo** [comment]
Have you tried following the [User Guide](https://github.com/DCS-Skunkworks/dcs-bios/blob/master/Scripts/DCS-BIOS/doc/userguide.adoc)?

**Reply by edwardker** [reply]
Hi, thank you for your reply. I understand how you connect 1 Led to 1 pin. But I have a pcb with 20 WS2812B leds where you can interpret them as a LEDstring with only 1 datapin to control 20 LEDs separate as each LED is seperatly adressable. So how do you define each LED in DCS-Bios? Thank you helping out

**Comment by talbotmcinnis** [comment]
You’d need to use the callbacks from dcs-bios and put your custom led code inside each callout.

I did something similar with my caution panel, here https://github.com/talbotmcinnis/McPit-DCS-BIOS/blob/6fb27a108dcd2a674379b780866dc9e5141ef3dd/McPit_Caution/McPit_Caution.ino

**Reply by edwardker** [reply]
Hi, thank you very much for sharing this. Much appreciated. I'm a noob so I am truly impressed with what you guys achieve.

**Comment by kbastronomics** [comment]
I have the code from OpenHornet that does just this but for the Caution Panel.  if that will help I'll post it here.

**Reply by edwardker** [reply]
Wow, thank you very much. It is the same problem for me with that panel. That would be very helpful. Thank you so much.

**Comment by kbastronomics** [comment]
[Caution Panel.txt](https://github.com/user-attachments/files/17851521/Caution.Panel.txt)

rename to .ino and make sure you have the adafruit  neopixel library installed

**Reply by edwardker** [reply]
Hi, thank you very, very much. This is amazing. I would never have found how to do this. Respect to what you can do. Thanks again!!! Could this run standalone on an Arduino?

**Comment by kbastronomics** [comment]
set #define ws2812_pin      A3  to the pin you're using
this assumes only the caution panel is hooked up and that there are 2 LEDS per indictor

**Reply by edwardker** [reply]
This is amazing!!!! Thank you very very much!!!! I would not in a million years been able to write this. Thanks a lot!!!
